frontend/workflows/projectSelector/src/selector-reducer.tsx

```diff
@@ -49,43 +51,48 @@ const selectorReducer = (state: State, action: Action): State => {
     case "ADD_PROJECTS": {
       let newState = { ...state };
 
-      // a given custom project may already exist in the group so don't trigger a state update for those duplicates
+      // a given custom project may already exist in the group so don't trigger a state update for the duplicate(s)
       const uniqueCustomProjects = action.payload.projects.filter(
-        (project: string) => !(project in newState[action.payload.group])
+        (project: string) => !(project in state[action.payload.group])
```

Comment:
thought it might be better practice to check the original state instead of the copied state